File: OO_design_patterns_training_2022-11/Solutions/C#/Exercise/DataSourceAdapter.cs
using OverlyIntricateDataSource;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Exercise
{
    class DataSourceAdapter
    {
        private readonly OverlyIntricateDataGenerator m_dataGenerator;

        public DataSourceAdapter()
        {
            m_dataGenerator = CreateDataSource();
        }

        public List<string> GetData()
        {
            return m_dataGenerator.CreateSeptupleTemporalSpacing().ToList();
        }

        private static OverlyIntricateDataGenerator CreateDataSource()
        {
            var dataSource = new OverlyIntricateDataGenerator();
            dataSource.Initialize(DateTime.Now.ToLongDateString().GetHashCode().ToString());
            dataSource.SetDataGenerationComplexity(DataGenerationComplexity.Intricate);
            dataSource.SetDataGenerationSpeed(DataGenerationSpeed.Mega);
            return dataSource;
        }
    }
}

File: OO_design_patterns_training_2022-11/MySolutions/C++/Exercise/src/DataSourceAdapter.cpp
#include "DataSourceAdapter.h"

#include <chrono>

using DataGenerator::DataGenerationComplexity;
using DataGenerator::DataGenerationSpeed;
using DataGenerator::OverlyIntricateDataGenerator;

std::vector<std::string> DataSourceAdapter::GetData() const
{
    return _dataGenerator.CreateSeptupleTemporalSpacing();
}

OverlyIntricateDataGenerator DataSourceAdapter::CreateDataSource()
{
    OverlyIntricateDataGenerator dataSource;
    dataSource.Initialize(std::to_string(std::chrono::system_clock::to_time_t(std::chrono::system_clock::now())));
    dataSource.SetDataGenerationComplexity(DataGenerationComplexity::Intricate);
    dataSource.SetDataGenerationSpeed(DataGenerationSpeed::Mega);

    return dataSource;
}